### ISSUE TITLE
Issue with ESPHome 2023.9.0

### ISSUE THREAD

**Issue by golles** [Sanderhuisman/ESPHome-Zehnder-RF#4]
I've to admit, that I haven't debugged it myself yet as I don't speak C, so that makes it a bit hard. 
I do have collected a bit of data, including potential relevant tickets in the bottom of this issue.

If you want me to test something out, let me know, as I'm happy to do so!
And thanks again for this project, I've been using it for a very long time with a lot of pleasure!

## Logs:
```
[I][logger:268]: Log initialized
[C][ota:473]: There have been 0 suspected unsuccessful boot attempts.
[D][esp32.preferences:114]: Saving 1 preferences to flash...
[D][esp32.preferences:143]: Saving 1 preferences to flash: 0 cached, 1 written, 0 failed
[I][app:029]: Running through setup()...
[D][spi:039]: Setting up SPI bus...
[D][binary_sensor:034]: 'Status': Sending initial state OFF
[D][text_sensor:064]: 'Version': Sending state '2023.9.0 Sep 27 2023, 17:25:16'
[C][zehnder:089]: ZEHNDER 'Ventilatie':
[E][spi:078]: SPIDevice not initialised - did you call spi_setup()?
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x40174e83  PS      : 0x00060630  A0      : 0x80174d66  A1      : 0x3ffb2510  
A2      : 0x3ffba708  A3      : 0x3ffb2566  A4      : 0x0000000b  A5      : 0x3ffb253c  
A6      : 0x3ffb2510  A7      : 0x00000008  A8      : 0x80174e9e  A9      : 0x3ffb24f0  
A10     : 0x00000000  A11     : 0x00000001  A12     : 0x3ffb2566  A13     : 0x0000000b  
A14     : 0x0000000b  A15     : 0x00000000  SAR     : 0x00000004  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x00000000  LBEG    : 0x4008a0a1  LEND    : 0x4008a0b1  LCOUNT  : 0xffffffec  


Backtrace:0x40174e80:0x3ffb25100x40174d63:0x3ffb2530 0x400db000:0x3ffb2550 0x400db073:0x3ffb25a0 0x400e17aa:0x3ffb25c0 0x40175359:0x3ffb2670 0x40175409:0x3ffb2690 0x400e2764:0x3ffb26b0 0x400e54b2:0x3ffb26e0 0x400f1616:0x3ffb2820 




ELF file SHA256: 0000000000000000

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13132
load:0x40080400,len:3036
entry 0x400805e4
```

## Config
See https://github.com/golles/ESPHome-Config/blob/main/mechanische_ventilatie.yaml

## Related
2023.9 release notes: https://github.com/esphome/esphome/releases/tag/2023.9.0
Changes related to fan:
- https://github.com/esphome/esphome/pull/5325

Changes related to spi:
- https://github.com/esphome/esphome/pull/5311
- https://github.com/esphome/esphome/pull/5416

Maybe similar:
- https://github.com/esphome/issues/issues/4911

**Comment by renoutg** [comment]
Same here, can confirm that after downgrading to esphome 2023.8.3 it works again

**Comment by golles** [comment]
https://github.com/esphome/esphome/pull/5446/ has fixed this issue and was released in [2023.9.2](https://github.com/esphome/esphome/releases/tag/2023.9.2)
